civo/acceptance/acceptance.go

```diff
@@ -0,0 +1,84 @@
+package acceptance
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"testing"
+
+	"github.com/civo/terraform-provider-civo/civo"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// TestAccProvider is a global instance of the provider under test.
+// It is used in acceptance tests to configure resources.
+var TestAccProvider *schema.Provider
+
+// TestAccProviders is a map of provider instances keyed by their name.
+// It is used in acceptance tests where multiple providers are in play.
+var TestAccProviders map[string]*schema.Provider
+
+// TestAccProviderFactories is a map of functions that return a provider instance and an error.
+// It is used in acceptance tests where the provider needs to be configured in a certain way.
+var TestAccProviderFactories map[string]func() (*schema.Provider, error)
+
+func init() {
+	TestAccProvider = civo.Provider()
+	TestAccProviders = map[string]*schema.Provider{
+		"civo": TestAccProvider,
+	}
+	TestAccProviderFactories = map[string]func() (*schema.Provider, error){
+		"civo": func() (*schema.Provider, error) {
+			return TestAccProvider, nil
+		},
+	}
+}
+
+// TestProvider - Test the provider itself
+func TestProvider(t *testing.T) {
```

Comment:
the tests should be inserted into `<namefile>_test.go` file, else are not executed.

Reply:
No, this one is fine, it is just and utility

Reply:
IMO test utilities should return objects and function for tests, not having tests inside. I trust your expertise.